OpenRA.Mods.Common/Widgets/TextFieldWidget.cs

```diff
@@ -115,82 +120,183 @@ public override bool HandleKeyPress(KeyInput e)
 			if (!HasKeyboardFocus)
 				return false;
 
-			if ((e.Key == Keycode.RETURN || e.Key == Keycode.KP_ENTER) && OnEnterKey())
-				return true;
+			Func<int> getPrevWhitespaceIndex = () =>
```

Comment:
I don't see any particular need to have these defined internally, so you should pull them out into helper methods so that they aren't reallocated every key event.